bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -914,6 +914,15 @@ impl Connection {
         }?;
         Ok(self)
     }
+
+    /// Allows the quic library to check if session tickets are expected
+    pub fn client_resumption_enabled(&self) -> bool {
+        if let Some(config) = self.config() {
+            let ctx = config.context();
+            return ctx.session_ticket_callback.is_some();
+        }
```

Comment:
Is this safe? Our C code won't trigger the callback if tickets aren't enabled on the config: https://github.com/aws/s2n-tls/blob/main/tls/s2n_server_new_session_ticket.c#L338-L340 The bindings enable tickets automatically: https://github.com/aws/s2n-tls/blob/main/bindings/rust/s2n-tls/src/config.rs#L511 But the low level bindings API is still a toggle, so someone could turn it off: https://github.com/aws/s2n-tls/blob/main/api/s2n.h#L1199

Reply:
What do you mean exactly with "is this safe"? What scenario are you worried about?

Reply:
Changed it so that we are checking both the resumption flag and the rust callback.

Reply:
In that case couldn't the low level bindings be used to set a callback, and then this would also not see it? Should all of the logic just be in C so `use_tickets` and the actual callback can be checked? Or is setting a binding outside of the actual bindings not really a concern?

Reply:
Hee hee yeah I changed it. It's cleaner code anyways.